manager/src/main/java/org/openremote/manager/dashboard/DashboardResourceImpl.java

```diff
@@ -202,19 +191,31 @@ protected DashboardQuery createDashboardQuery(String realm) {
 
         // If not logged in, force only public read/write access
         else {
-            userViewAccess = new HashSet<>(Set.of(DashboardAccess.PUBLIC));
-            userEditAccess = new HashSet<>(Set.of(DashboardAccess.PUBLIC));
+            userAccess = new HashSet<>(Set.of(DashboardAccess.PUBLIC));
             assetAccess = new HashSet<>(Set.of(DashboardQuery.AssetAccess.REALM));
         }
 
         // Build query object and return
         return query.conditions(new DashboardQuery.Conditions(
                         new DashboardQuery.DashboardConditions()
-                                .viewAccess(userViewAccess.toArray(new DashboardAccess[0]))
-                                .editAccess(userEditAccess.toArray(new DashboardAccess[0])),
+                                .access(userAccess.toArray(new DashboardAccess[0])),
                         new DashboardQuery.AssetConditions()
                                 .access(assetAccess.toArray(new DashboardQuery.AssetAccess[0]))
                 )
         );
     }
+
+    /**
+     * Function that builds an {@link DashboardQuery} object, based on user permissions.
+     * Automatically fills NULL values, adjusts access filters based on roles, etc.
+     *
+     * @param realm The realm to create a DashboardQuery for.
+     * @return An instantiated DashboardQuery with the correct permissions based on the user.
+     */
+    protected DashboardQuery createDashboardQuery(String realm) {
```

Comment:
This method seems un-necessary as `fixDashboardQuery` is called from here anyway and also that method sets the realm if it is not set